src/s3_backend.ts

```diff
@@ -0,0 +1,104 @@
+import fs from "fs";
+import { PullRequestInfo } from "./gh_helper";
+import { TerraformBackend } from "./tfc_cli";
+import { CommandVars } from "./comment_parser";
+import { 
+    S3Client, 
+    PutObjectCommand, 
+    GetObjectCommand,
+    DeleteObjectCommand
+} from "@aws-sdk/client-s3";
+
+export class S3Backend implements TerraformBackend {
+    private readonly workspaceName: string;
+    private readonly bucketName: string; 
+    private readonly s3Client: S3Client;
+
+    constructor(
+        workspaceName: string
+    ) {
+        this.bucketName = "lcv-tfstate"
+        this.workspaceName = workspaceName;
+        this.s3Client = new S3Client()
+    }
+    
+    public async cleanUp(): Promise<boolean> {
```

Comment:
here we clean up any state remnants from the s3 bucket after destroying all the resources.